plugins/genesys/fu-genesys-hubhid-device.c

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright (C) 2023 Adam.Chen <Adam.Chen@genesyslogic.com.tw>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-genesys-common.h"
+#include "fu-genesys-hubhid-device.h"
+
+#define GENESYS_HUBHID_REPORT_ID	  0
+#define GENESYS_HUBHID_REPORT_BYTE_LENGTH 0x40
+#define GENESYS_HUBHID_REPORT_TIMEOUT	  100 /* ms */
+#define GENESYS_HUBHID_REPORT_FLAGS	  FU_HID_DEVICE_FLAG_ALLOW_TRUNC | FU_HID_DEVICE_FLAG_IS_FEATURE
+
+struct _FuGenesysHubhidDevice {
+	FuHidDevice parent_instance;
+	GByteArray *buf_report;
```

Comment:
it seems a bit weird to put this as device instance data... it seems like the kind of thing that should be returned from fu_genesys_hubhid_device_command_read()

Reply:
The `buf_report` is used to send/receive GL report, and the size should be queried by HID report descriptor(the report size must fixed as the descriptor declared). Currently I set it a hardcode value to go.

Reply:
I think storing the size makes sense, just much less the buffer. It's acting as a cache.